File: t/scala/TestClass.scala

class TestClass() {
    private val _fsm = new TestClassContext(this)
    var myProp: Boolean = false

    // Uncomment to see debug output.
    // _fsm.setDebugFlag(true)

    def NoArg() {
        System.out.println("No arg")
    }

    def Output(str: String) {
        System.out.println(str)
    }

    def Output_n(str1: String, n: Int, str2: String) {
        System.out.print(str1)
        System.out.print(n)
        System.out.println(str2)
    }

    def isOk(): Boolean = {
        return true
    }

    def isNok(): Boolean = {
        return false
    }

    def Evt_1() {
        _fsm.Evt_1()
    }

    def Evt_2() {
        _fsm.Evt_2()
    }

    def Evt_3() {
        _fsm.Evt_3()
    }

    def Evt1(n: Int) {
        _fsm.Evt1(n)
    }

    def Evt2(n: Int) {
        _fsm.Evt2(n)
    }

    def Evt3(n: Int) {
        _fsm.Evt3(n)
    }

}

File: t/php/TestClass.php

<?php

require_once 'TestClassContext.php';

class TestClass {

    protected $_fsm;

    public function __construct() {
        $this->_fsm = new TestClass_sm($this);
        $this->_is_acceptable = false;

        // Uncomment to see debug output.
        //$this->_fsm->setDebugFlag(true);
    }

    public function getFSM() {
        return $this->_fsm;
    }

    public function NoArg() {
        echo "No arg\n";
    }

    public function Output($str) {
        echo "$str\n";
    }

    public function Output_n($str1, $n, $str2) {
        echo "${str1}${n}${str2}\n";
    }

    public function isOk() {
        return true;
    }

    public function isNok() {
        return false;
    }

    public function Evt_1() {
        $this->_fsm->Evt_1();
    }

    public function Evt_2() {
        $this->_fsm->Evt_2();
    }

    public function Evt_3() {
        $this->_fsm->Evt_3();
    }

    public function Evt1($n) {
        $this->_fsm->Evt1($n);
    }

    public function Evt2($n) {
        $this->_fsm->Evt2($n);
    }

    public function Evt3($n) {
        $this->_fsm->Evt3($n);
    }
}
?>
